crates/codegen/runtime/cargo/src/runtime/napi_interface/diagnostic.rs

```diff
@@ -0,0 +1,60 @@
+use napi_derive::napi;
+
+use crate::napi_interface::text_index::TextRange;
+
+/// Severity of the compiler diagnostic.
+///
+/// Explicitly compatible with the LSP protocol.
+#[napi(namespace = "diagnostic")]
```

Comment:
IIRC, we decided on using `napi(string_enum)` everywhere for debuggability. I understand that the LSP protocol is using numbers for performance, but I think this will be less of a concern on the user side, given that they are limited and easily convertible?
WDYT?

Reply:
If we want to have a Slang-specific enum we could use a string_enum, but if we're aiming for the LSP compatibility, then that's a no-go.

I'd like to limit proliferation of types and unify as much as possible and I don't think this value will be used when debugging, i.e. the only time someone will care if it's a warning or not is when we will write the LSP bridge and once that's implemented, it's hard to imagine that that the diagnostic severity levels will be mixed up somehow.

Reply:
https://github.com/NomicFoundation/slang/pull/963#discussion_r1611827890

Happy to have a quick f2f discussion about it if you have a different opinion.